File: 卡码/代码随想录/1/03-哈希表/Java/src/_02_两个数组交集/LeetCode_0349.java
//349. 两个数组的交集
package _02_两个数组交集;

import java.util.HashSet;

public class LeetCode_0349 {

}
class Solution {
  public int[] intersection(int[] nums1, int[] nums2) {
  	HashSet<Integer> hash = new HashSet<Integer>();
  	HashSet<Integer> res = new HashSet<Integer>();
  	for(int num:nums1) hash.add(num);
  	for(int num:nums2)
  		if(hash.contains(num)) res.add(num);
  	return res.stream().mapToInt(num->num).toArray();
  }
}

File: 卡码/代码随想录/C++/哈希/02-两个数组交集/LeetCode_0349.cpp
//349. 两个数组的交集
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
  vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
		unordered_set<int> hash;
		unordered_set<int> res;
		for(int num:nums1) hash.insert(num);
		for(int num:nums2)
			if(hash.find(num)!=hash.end()) res.insert(num);
		return vector<int>(res.begin(),res.end());
  }
};
